src/backend/cuda/kernel/ireduce.hpp

```diff
@@ -22,11 +22,30 @@ namespace cuda
 {
 namespace kernel
 {
+    template<typename T> __host__ __device__ 
+    static double cabs(const T& in) { return (double)in; }
```

Comment:
@pavanky I am questioning the reasoning behind using doubles as the output of this function. Integral types should return the same type. I understand wanted to keep the original C return value but we are leaving a lot of performance on the table here. @mark-poscablo I am making general comments. You don't need to change anything in this PR.